### PR TITLE
Dev/location test pr

### PR DESCRIPTION
**Pull request by miooo0o**
Implemented the import of Config and Location classes and the creation of the correct response.
- resolve build errors due to circular dependency
- modularization of existing methods, logic refined in the meeting with @san-ghun 
- documentation
- Renamed/unified some methods (mostly in HttpResponse, StaticFileHandler files)
- Change `ext` in `mime-type` list  so that it does not contain `.`

tested:
```
http://localhost:8080/fruits/index.html    -------> OK (file opened)
http://localhost:8080/fruits/   ------------------> OK (index.html fetching)
http://localhost:8080/    ------------------------> OK (index.html fetching)
```

### CODE REVIEW

**Review by san-ghun — CHANGES_REQUESTED** [review]
Hey @miooo0o ,
thanks for you hard work.

I see some parts that need modification.
Please have look at my review and change those points.

Leave comment or message after your modification.

Thanks!

File: include/RequestHandler.hpp [hunk 1]
```diff
@@ -6,27 +6,42 @@
...
+class Location;
+
+# include <algorithm>
 # include "StaticFileHandler.hpp"
+# include "webserv.hpp"
 
+std::string _extractPathFromUri(const std::string& uri);
+std::string _getMatchedLocation(std::string path,
+								const std::map<std::string,
+								Location*>& locations);
```
[the remark this answers is not in the document]
Comment (miooo0o) [reply include/RequestHandler.hpp:27]: 함수 위치가 확정이 안되서요. 따로 변경사항 없으면 나중에 member로 넣을거에요.

File: src/server/StaticFileHandler.cpp [hunk 2]
```diff
@@ -3,93 +3,287 @@
...
+	if (!file.read(&body[0], fileLength))
+	{
+		*this = internalServerError_500();
+		return ("");
+	}
+ */
+/// @brief Generates an HTML page with a directory listing.
+/// @param path 
+/// @return Create a list of directories and files in HTML and export it to std::stringstream.
+std::string	genDirListingHtml(const std::string& path)
```
[the remark this answers is not in the document]
Comment (miooo0o) [reply src/server/StaticFileHandler.cpp:154]: 오 그것도 좋은 생각입니다! 사실 이 함수도 일단은 member로 넣을 이유가 크게 없어서 빼둔건데 계속 유지하는거로 생각할게요!

File: src/server/StaticFileHandler.cpp [hunk 3]
```diff
@@ -3,93 +3,287 @@
...
 {
 	if (request.getUri() == "/")
-		return (_handleRoot());
+		return (_handleRoot(location));
+	_setHandledPath(_buildPathWithUri(request, location));
+	if (isDir(_handledPath))
+	{
+		// if (/* && location.location.isListDir() */)
+		// 	return (_handleDirListing(request, location));
+	                                                                                                                                                                                                                                                                                      	return (_handleDirRequest(request, location));
```
[the remark this answers is not in the document]
Comment (miooo0o) [reply src/server/StaticFileHandler.cpp:67]: 확인하고 수정할게요!

File: src/server/StaticFileHandler.cpp [hunk 4]
```diff
@@ -3,93 +3,287 @@
...
+HttpResponse StaticFileHandler::_handleDirRequest(const HttpRequest& request, const Location& location)
+{
+	HttpRequest	modifiedRequest = request;
+
+	std::string absolutePath = _buildAbsolutePathWithIndex(location);
+	_setHandledPath(absolutePath);
+
+	std::cout << "TEST |  absolute path that with index is built: " << _handledPath << std::endl;
+	if (isFile(_handledPath))
+		return (_handleNotFound());
```
[the remark this answers is not in the document]
Comment (miooo0o) [reply src/server/StaticFileHandler.cpp:92]: 이런 치명적인 실수를. 감사합니다, 수정할게요.

**Review by san-ghun — APPROVED** [review]
Thanks for your correction.

**Pull request merged**